whirlpool/appliance.py

```diff
@@ -35,20 +37,24 @@ def __init__(
         self._session: aiohttp.ClientSession = session
 
     def register_attr_callback(self, update_callback: Callable):
+        """Register Callback function."""
         self._attr_changed.append(update_callback)
         LOGGER.debug("Registered attr callback")
 
     def unregister_attr_callback(self, update_callback: Callable):
-        try:
-            self._attr_changed.remove(update_callback)
-            LOGGER.debug("Unregistered attr callback")
-        except ValueError:
-            LOGGER.error("Attr callback not found")
+        """Unregister callback function."""
+        if self._attr_changed:
```

Comment:
there is no way this can be None so I don't think this check is needed

Reply:
It can be, if you unregistered multiple times it may be None, if we don't test for it, it raises an error.

Reply:
It is never set to None. If you mean unregister being called when the array is empty, then the "Attr callback not found" error should indeed be logged, because that should not happen and the client (HA?) needs to fix that.